atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1872,11 +1902,33 @@ public void commit() {
 
     @Override
     public void commit(TransactionService transactionService) {
+        runPreCommitCallbacks();
         commitWithoutCallbacks(transactionService);
         runSuccessCallbacksIfDefinitivelyCommitted();
         close();
     }
 
+    private void runPreCommitCallbacks() {
+        preCommitActions.forEach((lockDescriptor, actions) -> {
```

Comment:
@ergo14 this is where your RPC would come in; @fsamuel-bs should we try to be able to batch across all of the named timestamps?